lib/wallet/rpc.js

```diff
@@ -1402,34 +1413,65 @@ class RPC extends RPCBase {
   }
 
   async sendToAddress(args, help) {
-    if (help || args.length < 2 || args.length > 5) {
-      throw new RPCError(errs.MISC_ERROR,
-        'sendtoaddress "address" amount'
-        + ' ( "comment" "comment-to" subtractfeefromamount )');
-    }
+    const opts = this._validateSendToAddress(args, help, 'sendtoaddress');
+    const wallet = this.wallet;
 
+    const options = {
+      subtractFee: opts.subtract,
+      outputs: [{
+        address: opts.addr,
+        value: opts.value
+      }]
+    };
+
+    const tx = await wallet.send(options);
+
+    return tx.txid();
+  }
+
+  async createSendToAddress(args, help) {
+    const opts = this._validateSendToAddress(args, help, 'createsendtoaddress');
     const wallet = this.wallet;
+
+    const options = {
+      paths: true,
+      account: opts.account,
+      subtractFee: opts.subtract,
+      outputs: [{
+        address: opts.addr,
+        value: opts.value
+      }]
+    };
+
+    const mtx = await wallet.createTX(options);
+
+    return mtx.getJSON(this.network);
+  }
+
+  _validateSendToAddress(args, help, method) {
+    const msg = `${method} "address" amount `
+      + '( "comment" "comment-to" subtractfeefromamount "account" )';
```

Comment:
I notice that this has additional arguments (`comment`, `comment-to`) that are meant to keep it compatible with `bitcoind`. These arguments are not used.

https://bitcoincore.org/en/doc/0.17.0/rpc/wallet/sendtoaddress/

Adding `account` breaks the compatibility as `replaceable` is supposed to follow `subtractfeefromamount`.
I think that `account` should replace `comment` so that we can maintain compatibility, and then we have 1 more slot `comment-to` for something `hsd` specific.

Reply:
We may want to support "comment" and "comment-to" eventually. Also, `replaceable` is a BIP125 specific argument. Since Handshake does not support BIP125, I prefer to favor backward compatibility with our current codebase, vs. compatibility with the `bitcoind` RPC interface. At least for `hsd`.